kolibri/core/auth/management/commands/importusers.py

```diff
@@ -24,90 +28,72 @@ def validate_username(user):
     username = user_dict["username"]
-    demo_fields = ["gender", "birth_year", "id_number"]
-    user_updated = any(user_dict.get(key, None) is not None for key in demo_fields)
+    user_updated = any(user_dict.get(key, None) is not None for key in DEMO_FIELDS)
 
     if not user_updated:
         return
 
+    user_updated = False
+
     if user_dict.get("gender", None) is not None:
-        user_model.gender = user_dict["gender"]
+        value = transform_inputs("gender", user_dict)
+        user_updated = user_updated or value != user_model.gender
+        user_model.gender = value
 
     if user_dict.get("birth_year", None) is not None:
-        user_model.birth_year = user_dict["birth_year"]
+        value = transform_inputs("birth_year", user_dict)
+        user_updated = user_updated or value != user_model.birth_year
+        user_model.birth_year = value
 
     if user_dict.get("id_number", None) is not None:
-        user_model.id_number = user_dict["id_number"]
+        value = user_dict["id_number"]
+        user_updated = user_updated or value != user_model.id_number
+        user_model.id_number = value
 
     try:
-        user_model.full_clean()
-        user_model.save()
-        logger.info(
-            'User "{username}" was updated with demographic info'.format(
-                username=username
+        if user_updated:
+            user_model.full_clean()
+            user_model.save()
+            logger.info(
+                'User "{username}" was updated with demographic info'.format(
+                    username=username
+                )
             )
-        )
 
     except ValidationError as e:
         logger.error(
-            'Tried to update demogrpahic info for "{username}", but invalid values were given for fields: {keys}'.format(
+            'Tried to update demographic info for "{username}", but invalid values were given for fields: {keys}'.format(
                 username=username, keys=list(e.error_dict.keys())
             )
         )
 
 
-def create_user(i, user, default_facility=None):
+def create_user(user, default_facility=None):
     validate_username(user)
 
-    if i == 0 and all(key == val or val is None for key, val in user.items()):
-        # Check whether the first row is a header row or not
-        # Either each key will be equal to the value
-        # Or the header is not included in the CSV, so it is None
-        return False
-
-    facility = infer_facility(user, default_facility)
-    classroom = infer_and_create_class(user, facility)
+    facility = infer_facility(user.get("facility", None), facility=default_facility)
+    classroom = infer_and_create_class(user.get("class", None), facility)
     username = user["username"]
     try:
         user_obj = FacilityUser.objects.get(username=username, facility=facility)
         update_user_demographics(user, user_obj)
 
-        if classroom:
+        if classroom and not user_obj.is_member_of(classroom):
```

Comment:
I had one user in the export already in a classroom, but they fell through into this branch and got re-added into classroom. But there was another user (different classroom) that didn't go into this branch. Not sure what the difference is between the two though.

Reply:
Or at least the export gave them a 'classroom_id' in the CSV. Maybe they weren't actually in the classroom, hence they went into this branch.

Reply:
Okay, the problem seems to be that the CSV export is giving me these rows. Notice they are all for user `jb` (I need to test if it wil cause problems if I edit one or not the others, give them different, data, etc.)

The thing I'm noticing is that the first row for class "jb/...38c2" is that I don't actually see that Classroom when I run the app. I'm not sure what that's referring to or if that's pollution from a previous install.

```
Facility name,Facility id,Class name,Class id,Full name,Gender,Birth year,ID number,Username,Password
a,6f963c47fefcfbe2d1aaec8490dba5da,jb,439ab8832b5b11d071204fd8aa5e38c2,jb,,,,jb,
a,6f963c47fefcfbe2d1aaec8490dba5da,My Class,4fb98d1aef10ce4bb49f380a2f661c8b,jb,,,,jb,
a,6f963c47fefcfbe2d1aaec8490dba5da,jajajaa,cbc3d5e53eaf3ea6c9e57ba59b394516,jb,,,,jb,

```

Reply:
Ah, I failed to document this, but this was the easiest way to manage a learner in multiple classes (rather than dynamically adding additional headers for each class, and with no way of preferentially selecting which class should be the one displayed).

Possible that the unidentified class is actually a group, so I'll double check the logic there.

Reply:
Confirmed, I failed to do that filter.

Reply:
I see. Maybe we can be more opinionated and assume that if we have the `-d` flag, then the primary goal is to display/edit demographic data. In that case, we could collapse multi-classroom users into a single row and have cell value like "Multiple classrooms" for that column.

Reply:
Ok - will do this as standard, as I don't see much of a use case for having the multiple lines, except as a mechanism for showing how to put users in multiple classes at once.